### PR TITLE
School Library- Add unit test

### PR DESCRIPTION
**Pull request by Emmy-github-webdev**
# School Library- Add unit test

In this project, @Ntare22  and @Emmy-github-webdev 
- Add unit tests for Book, Person, Student, Teacher, classroom, Corrector, Rental

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  context 'Testing book class' do
+    it 'A book should have a title and author' do
+      book = Book.new('Title', 'Author')
+      title = book.title
+      author = book.author
+      book_title_author = [title, author]
+
+      expect(book_title_author).to eq %w[Title Author]
+    end
+  end
```
Comment (TedLivist) [inline spec/book_spec.rb]: - [ ] All class methods should be tested. Kindly add test for your `add_rental` method as contained in the **Book** class

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      id = rand(1..1000)
+      person = Person.new(21, 'Emmanuel', id)
+      person_info = [person.id, person.age, person.name, person.parent_permission]
+      expect(person_info).to eq [id, 21, 'Emmanuel', true]
+    end
+
+    it 'Can_use_service? Check age is greater than 18 years' do
+      person = Person.new(21)
+      expect(person.can_use_services?).to be true
+    end
```
Comment (TedLivist) [inline spec/person_spec.rb]: - [ ] Kindly add test for the `validate_name` method as contained in the **Person** class.

File: spec/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+describe Student do
+  context 'Testing student class' do
+    it 'A student should have a age and name' do
+      student = Student.new('Age', 'Name')
+      age = student.age
+      name = student.name
+      student_info = [age, name]
+
+      expect(student_info).to eq %w[Age Name]
+    end
```
Comment (TedLivist) [inline spec/student_spec.rb:13]: - [ ] Kindly add test for the `play_hooky` method.

File: spec/teacher_spec.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+require 'rspec'
+require_relative '../classes/teacher'
+
+describe Teacher do
+  context 'Testing teacher class' do
```
Comment (TedLivist) [inline spec/teacher_spec.rb:5]: - [ ] Kindly add test for the `can_use_services?` method in the Teacher class

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Emmy-github-webdev and @Ntare22,

Good job so far!
✔️ Tests are passing
✔️ Test files are contained in `spec` directory
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] **In your README kindly add instructions on how to run the tests as it can be of great use to users**
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/book_spec.rb (hunk 1 above)
Comment (Emmy-github-webdev) [reply spec/book_spec.rb]: Test added for add_rental

File: spec/person_spec.rb (hunk 2 above)
Comment (Emmy-github-webdev) [reply spec/person_spec.rb]: Test added for validate_name

File: spec/student_spec.rb (hunk 3 above)
Comment (Emmy-github-webdev) [reply spec/student_spec.rb:13]: Test for play_hooky added

File: spec/teacher_spec.rb (hunk 4 above)
Comment (Emmy-github-webdev) [reply spec/teacher_spec.rb:5]: Test for can_use_service in teacher class added

**Review by elyor-doniyorov — CHANGES_REQUESTED** [review]
Hi @Emmy-github-webdev and @Ntare22, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
- [ ] In the Readme file I think you forgot to add instructions on how to run the tests as the previous reviewer requested. Kindly add it.

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Emmy-github-webdev** [comment]
Thanks @elyor-doniyorov

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @Emmy-github-webdev , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

**Comment by bellom** [comment]
@Emmy-github-webdev - you seem to be working alone, and your PR description stated 2 people.

@TedLivist as a first reviewer, we need to change for equal commits.

**Comment by Emmy-github-webdev** [comment]
Thank you @bellom

**Pull request merged**